src/commands/copy/copy-rpc.ts

```diff
@@ -6,7 +6,7 @@ export interface CopyContractArgs {
   /** Copy ID for tracing */
   id: string;
   /** List of files that need to be copied */
-  manifest: { source: string; target: string }[];
+  manifest: { source: URL; target: URL }[];
```

Comment:
can worker threads actually pass URLS? I thought everything was meant to be a POJO for passing around?

Reply:
I assume you mean either JSON or JS object, not "[Plain old Java object](https://en.wikipedia.org/wiki/Plain_old_Java_object)". I have no idea how `WorkerRpc` works; I guess since you ask we don't have tests for this message passing?

Reply:
`worker_threads` are a mechanisim of using small scripts to add more threads to do processing, they are limited to posting messages to/from the thread using a rpc type mechanisim, `WorkerRpc` wraps the posting so it is somewhat typesafe.

I am not sure we can pass a URL between threads as its not really a easily searliazable thing, just like you cant pass a date but you can pass a ISO string.